api/http/handler/registry.go

```diff
@@ -312,3 +332,50 @@ func (handler *RegistryHandler) handleDeleteRegistry(w http.ResponseWriter, r *h
 		return
 	}
 }
+
+// validateRegistryURL validates wether given url is valid a docker registry url by
+// sending http get request to url using combination of protocols and available
+// registry versions. upon first successfull attempt, it returns protocol, version
+// and nil error.
+func validateRegistryURL(url string) (string, string, error) {
+	if !strings.HasSuffix(url, "/") {
+		url += "/"
+	}
+
+	configs := []struct {
+		protocol, version string
+	}{
+		{"https", "v2"},
+		{"https", "v1"},
+		{"http", "v2"},
+		{"http", "v1"},
+	}
+
+	client := &http.Client{
+		Timeout: registryCheckTimeout,
+	}
+	for _, config := range configs {
+		req, err := http.NewRequest("GET", fmt.Sprintf("%s://%s/%s/", config.protocol, url, config.version), nil)
+		if err != nil {
+			continue
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			continue
+		}
+
+		switch resp.StatusCode {
+		case http.StatusOK:
+		case http.StatusUnauthorized:
+		case http.StatusNotFound:
+			continue
+		default:
+			continue
+		}
```

Comment:
I don't understand this switch !
You have a valid registry URL only on 200 or 401 response

Reply:
good point, I also just realised it can be written better